### ISSUE TITLE
The DropListener gets called only for the first transaction

### ISSUE THREAD

**Issue by pablopantaleon** [braintree/braintree-android-drop-in#359]
### General information

* SDK/Library version: 6.2.0 
* Environment: SandBox 
* Android Version and Device: Android 10, Xiaomi Note 10 

### Issue description
The payment transaction only works the first time, the second time the DropListener is not called.

**Steps to Reproduce**
1. Use a Fragment
2. Trigger the `launchDropIn()`
3. Select a recent payment method
4. Complete the transaction
5. Open again the `launchDropIn()`
6. Select a recent payment method
7. Notice that the picker get closed but the DropListener is not called

I did some debug and the following gets called only the first time:

<img width="450" alt="Screen Shot 2022-08-17 at 01 24 45" src="https://user-images.githubusercontent.com/3310844/185059515-270200b7-a2c4-4633-9b78-7fc3c87e1c9a.png">

**Comment by sshropshire** [comment]
Hi @pablopantaleon thanks for using the Braintree SDK for Android. Can you share more information about how `DropInClient` is instantiated in your app? Also, are you able to skip step 4 and observe the same behavior?

**Comment by pablopantaleon** [comment]
@sshropshire sorry for the late reply and thanks for your help, it works. Skipping the step 4 solved the issue, so after that digging in more I found that when transaction completes I was doing:
```
supportFragmentManager.commit {
    remove(it)
}
```
in order to kill the current active fragment that holds the `DropInClient`, but that was causing the issue because after do that and open again the same fragment to process a new transaction, the `DropInClient` stops working. So I changed that to:
```
supportFragmentManager.popBackStack()
```
I'm not facing the issue anymore. Thanks.
